### ISSUE TITLE
Spell checking in Markdown cells

### ISSUE THREAD

**Issue by ylian** [HappenApps/Quiver#89]
Currently spell checking works in text cells, but not in Markdown cells.

The Ace editor does have an spell check extension. I will give it a try and see how well it works.

**Comment by papahabla** [comment]
Hope this worked well and is coming in future release! Very important feature, for me at least.

**Comment by ChristinWhite** [comment]
Did you have any luck working on this ylian? This would be my number one feature request with the new version.

**Comment by ylian** [comment]
No, haven't got time for this. Promise to work on this right after 3.0.

**Comment by ChristinWhite** [comment]
No rush, 3.0 is looking fantastic!

**Comment by smaant** [comment]
I would looove to have spell checking in Markdown cells, it's a real bummer without it now. Any news regarding this feature?

**Comment by jkjelsbak** [comment]
Please raise the attention on spell checking in Markdown cells

**Comment by acrodrig** [comment]
Same here. It would be very nice to have this feature.

**Comment by markgrovs** [comment]
+1

**Comment by yjlintw** [comment]
+1

**Comment by dkarter** [comment]
+1

**Comment by t-pare** [comment]
+1

**Comment by mhenry** [comment]
This would be great!

**Comment by LosMyke** [comment]
+1

**Comment by akaman** [comment]
+1

**Comment by danielkeil** [comment]
+1

**Comment by simon-johansson** [comment]
+1

**Comment by DenisLemos** [comment]
+1

**Comment by brettgoulder** [comment]
+1

**Comment by adamq-q** [comment]
+3, my whole team wants spellcheck so they can finally find happiness

**Comment by pelted** [comment]
NEEDED! Used the trial for 5 min before buying the app, then I realized there was no spell checking in Markdown cells. Please give this more attention.

**Comment by akaman** [comment]
Sadly, the lack of spell checking made me switch apps — after several months with Quiver.  I’ll try Quiver again if spell checking is added.

**Comment by acrodrig** [comment]
I already commented on this: I think it's a high priority. I am having to copy/paste to MS Word to spell check and back again. Not a nice workflow. Please do work on this.

**Comment by ylian** [comment]
My apologies. More details are here: #553.

I am in the process of rewriting the Ace editor to fix this (and many other editor-related issues).

**Comment by ChristinWhite** [comment]
I was playing with Ace recently, that's a big job! I can't wait to see the improvements!

**Comment by ntoombs19** [comment]
@ylian - This improvement would be very much appreciated. Nearly all my notes are in MD and my speling is attrocious ;)

**Comment by dlichtenberg** [comment]
+1 for spell check. @ylian, you've done a great job on the app. Proper spelling is so important for credibility in technical blog posts.  This feature is the biggest deciding factor for me on using the app.

**Comment by afulki** [comment]
Any progress on spell checking? I have pretty much started using Quiver for all my notes but having to copy the MD text to another editor for spellchecking is a PITA :(

**Comment by akaman** [comment]
I’m interested in an update also.  I’ve switched to Bear since that does MD and works well.  Not as feature rich as Quiver, but spelling is a deal-breaker.

**Comment by grstreten** [comment]
+1 - love Quiver, spellcheck in MD would be ace though!

**Comment by zachwood** [comment]
In the future, try not to spam with "+1" comments.  👍  Buttons were added to show interest in an issue.

**Comment by blablahaha** [comment]
It's already 2017, but still can not use spell check in markdown cell.

**Comment by darksheer** [comment]
This is a LONG time to have something like Spell check not functioning in an app like this.  Could we possibly get a new delivery date for this feature?

**Comment by johnpeach** [comment]
This would be a great app if the spell checker in MD worked.  While it is a limitation of ACE there are all kinds of replacements for ACE that support MD and spell checking. Not having spell checking makes the app I paid for look like amateur hour.  The fact that this is a highly requested (and basic feature) for the last 2+ years is just shameful.

**Comment by chrisjcameron** [comment]
For what it is worth, the open source editor MacDown has spell checking in Markdown that seems to use the System dictionary. Their code might be helpful: https://github.com/MacDownApp/macdown

**Comment by ylian** [comment]
@chrisjcameron That's very helpful, thanks.

**Comment by leejordan** [comment]
Hey @ylian I've been using quiver's free trial and I loved every aspect of it until I discovered spell check wasn't working in a markdown cell. I googled it and found this issue. I don't want to sound overly negative because I think you've made something great here but this is actually the one sticky point that has made me cautious about buying the software and I had to move back to macdown for writing markdown. If you can get macdown's spell checking working in quiver then I'll have no reservations about quiver at all.

I know it's hard to juggle user requests and bugs but I just wanted to flag that this is actually an important missing feature for me.

**Comment by 733amir** [comment]
+1

**Comment by paritosh16** [comment]
+1

**Comment by acrodrig** [comment]
Sadly, the lack of responsiveness on Quiver's side made move on to BearNotes. It's not as complete as Quiver form needs, but it's moving ahead, so hopefully it will catch up on the areas I need. 

@ylian Thanks for a great product.

**Comment by ylian** [comment]
@acrodrig If you are concerned about Quiver's future, don't be. I will be working full-time on Quiver again within a couple of months, and I am committed to working on Quiver for years to come.

**Comment by Piste** [comment]
Excellent news, @ylian !

**Comment by hunttom** [comment]
+1 It would be much appreciated.

**Comment by luispuerto** [comment]
+1 too!

**Comment by squk** [comment]
👍

**Comment by vlain** [comment]
+1

**Comment by soloradish** [comment]
+1

**Comment by neddy** [comment]
Just wanted to say that I am loving Quiver, but this is one thing I really wish it had... Every time I read my notes I find spelling errors...

**Comment by bemental** [comment]
Amazing app, would really like to see this functionality added.

**Comment by jayKlo** [comment]
Any timeline update on this? It's been nearly 4 years with no clear answer. I love Quiver but the lack of spell check has been a serious annoyance to my workflow.

**Comment by bemental** [comment]
I changed the default block to a text block and just ran with it since spell check works in those blocks. 

Isn’t ideal, but with the keyboard shortcut to change the block to markdown after I’m done typing, it works well enough.

**Comment by ylian** [comment]
@jayKlo I will have it fixed. Editor change is planned for Quiver 3.3, right after I push out 3.2 which is very close to release.

**Comment by jayKlo** [comment]
@ylian Fantastic! Looking forward to the update.

**Comment by akaman** [comment]
Finally!!!!!!!

**Comment by bemental** [comment]
Thanks for being a communicative and responsive dev, keep up the great work!

**Comment by bemental** [comment]
Thanks for another great update, getting closer to working on spellchecking in Markdown cells?

Happy holidays!

**Comment by ylian** [comment]
@bemental Yes, working on the editor rewrite right now. This is the main focus for Quiver 3.3.

**Comment by sanromd** [comment]
@ylian I have been using quiver for some months now. I really like it. I just wish the spell checker would work in markdown cells. any update on that front?
thank you

**Comment by ylian** [comment]
@sanromd Yes, editor rewrite is underway, so are several other major tasks.

**Comment by mdbergmann** [comment]
+1 for Markdown spell checking

**Comment by ruzz311** [comment]
@ylian - is there a road-map or real-time status of this rewrite?  Any chance of getting an update on the status of having spell-checking in markdown?

**Comment by ylian** [comment]
@ruzz311 The editor rewrite includes several things: a lossless Markdown parser, a syntax parser for code, the editor itself. So far the first two are done, and basic editor functionalities are working. This editor rewrite is high priority and I will get it done ASAP.

**Comment by randy** [comment]
@ylian Are you a one-man show on this? Or is there a repo where others can contribute?
